src/pushover_complete/pushover_api.py

```diff
@@ -68,20 +71,19 @@ def _generic_post(self, endpoint, url_parameter=None, payload=None, session=None
         if payload is None:
             payload = {}
         payload['token'] = self.token
-        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
```

Comment:
I'm curious: why was this here, and why is it no longer needed?

@scolby33 thoughts?

Reply:
The documentation for (Pushing Messages)[https://pushover.net/api#messages] specifies that Content-Type in the example. Sending without an image seemed to work whether it was there or not. When sending the image attachment though the Content-Type needs to be multipart/form-data and that is added automatically. I think it must be the default when specify the data payload.
If the headers includes the urlencoded Content-Type though, it overrides the form-data specification and causes the post to fail.

Reply:
I think I added this after doing some testing of the Pushover API with an external REST client that *didn't* add the `application/x-www-form-urlencoded` value. If Requests adds this by default as a guarantee, I'd love to see this line go. If not, we should have an if/else that puts the right header on depending on if there's an attachment or not. I should add sent header verification to the tests.

Reply:
I'll double check how the call performs with and without the file parameter, but I think the header is indeed not required here.

Reply:
I've tested with specifying a file and without and the `application/x-www-form-urlencoded` header is not needed.

Reply:
If it works, I'm happy to have this removed! I've opened #7 to remind myself to add tests later just in case `requests` changes it's behavior someday.